src/main/java/fi/aalto/cs/apluscourses/ui/exercise/ExercisesTreeRenderer.java

```diff
@@ -5,12 +5,29 @@
 import fi.aalto.cs.apluscourses.presentation.exercise.ExerciseViewModel;
 import fi.aalto.cs.apluscourses.presentation.exercise.ExercisesTreeViewModel;
 import icons.PluginIcons;
+import javax.swing.Icon;
 import javax.swing.JTree;
 import javax.swing.tree.DefaultMutableTreeNode;
 import org.jetbrains.annotations.NotNull;
 
 public class ExercisesTreeRenderer extends ColoredTreeCellRenderer {
 
+  @NotNull
+  private static Icon statusToIcon(@NotNull ExerciseViewModel.Status exerciseStatus) {
+    switch (exerciseStatus) {
+      case NO_SUBMISSIONS:
+        return PluginIcons.A_PLUS_NO_SUBMISSIONS;
+      case NO_POINTS:
+        return PluginIcons.A_PLUS_NO_POINTS;
+      case PARTIAL_POINTS:
+        return PluginIcons.A_PLUS_PARTIAL_POINTS;
+      case FULL_POINTS:
+        return PluginIcons.A_PLUS_FULL_POINTS;
+      default:
+        return null;
```

Comment:
`@NotNull`?

Reply:
This should never happen, but you're right, this is silly, I'll throw `IllegalStateException`

Reply:
`throw null` :wink:

Reply:
`IllegalArgumentException` is perhaps more accurate.